hugr-passes/src/replace_types.rs

```diff
@@ -109,28 +130,42 @@ impl NodeTemplate {
 
     fn replace(self, hugr: &mut impl HugrMut<Node = Node>, n: Node) -> Result<(), BuildError> {
         assert_eq!(hugr.children(n).count(), 0);
-        let new_optype = match self {
-            NodeTemplate::SingleOp(op_type) => op_type,
+        let (new_optype, static_source, static_inport) = match self {
+            NodeTemplate::SingleOp(op_type) => (op_type, None, None), // perhaps assert op_type has no static input?
```

Comment:
Static inputs in a single op would never be valid since the signature wouldn't match up with the one we're replacing, right?